### DISCUSSION TITLE
Keycloak get dynamic hostname path from proxy

### DISCUSSION THREAD

**Discussion by ghost** [keycloak/keycloak#19091] [Q&A]
### Description

I use nginx reverse proxy for Keycloak

Keycloak configuration: 

- hostname-strict=false
- http-relative-path not set
- hostname-path not set
- port 8080

nginx configuration:
``` nginx
listen       8081;
server_name  localhost;
location /auth/ {
            proxy_pass http://localhost:8080/;
            proxy_set_header Host localhost:8081;
            proxy_set_header Path auth;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Prefix auth;
}
```

I access the browser: http://localhost:8081/auth/admin, The address redirect to http://localhost:8081/admin/master/console/

Currently latest version, does Keycloak support dynamic path parsing?

**http-relative-path** I have tested it and can use it.
**hostname-path** I have tested it and only realm api, resources path can't parse.

Production environment, I have deployed a kubernetes cluster, I need multiple domain names to access this cluster, and Some sites need sub prefix such http://example.org/auth.  Is there any best practice in this scenario? I want to use a database and deploy multiple clusters, ISPN is only shared within the cluster, Is this OK? thank you!

**Comment by sschu** [comment]
You can configure Keycloak to run under a different path by setting for example `KC_HTTP_RELATIVE_PATH=/auth`. But this is for the whole instance.

**Reply by MartinWeise** [reply]
Thank you so much, I was struggling a lot with reverse proxy config for days and this finally solved it!
